src/main/java/com/chopshop166/chopshoplib/leds/LEDStripBuffer.java

```diff
@@ -0,0 +1,126 @@
+package com.chopshop166.chopshoplib.leds;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import edu.wpi.first.wpilibj.AddressableLED;
+import edu.wpi.first.wpilibj.AddressableLEDBuffer;
+
+public class LEDStripBuffer {
+    private List<SegmentConfig> segmentConfigs = new ArrayList<>();
+    private Map<String, Set<SegmentConfig>> segmentTagMap = new HashMap<>();
+    private Map<SegmentConfig, Pattern> patternMap = new HashMap<>();
+    private List<RunOrder> runOrder = new ArrayList<>();
+    private Set<Pattern> newPatterns = new HashSet<>();
+    private int numLEDs;
+    private int nextLED;
+    private final AddressableLEDBuffer buffer;
+
+    public LEDStripBuffer(int numLEDs) {
+        this.numLEDs = numLEDs;
+        this.nextLED = 0;
+        buffer = new AddressableLEDBuffer(numLEDs);
+    }
+
+    public int getLength() {
+        return numLEDs;
+    }
+
+    public SegmentConfig segment(int length) {
+        assert numLEDs >= nextLED + length;
+        assert length > 0;
+        var result = new SegmentConfig(this);
+        result.startPos = nextLED;
+        result.stopPos = nextLED + length - 1;
+        result.length = length;
```

Comment:
This seems like it could be a constructor that takes `buffer`, `startPos`, and `length`.